### PR TITLE
{Pylint} Fix redundant-u-string-prefix

### PR DESCRIPTION
**Pull request by atombrella**
**Description**

Another bit of legacy from the past Python 2 days. https://github.com/Azure/azure-cli/pull/20192 deactivated the particular u-prefix linter.

There are several instances of Python 2 import-guards. I fixed a couple of them in affected files. Maybe those are unaffected. As far as I can see, Python 3.5 is a minimum requirement. 

**History Notes**

This checklist is used to make sure that common guidelines for a pull request are followed.

- [X] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [X] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [X] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

File: src/azure-cli/azure/cli/command_modules/keyvault/vendored_sdks/azure_keyvault_securitydomain/_serialization.py [hunk 1]
```diff
@@ -931,13 +928,6 @@ def serialize_unicode(cls, data):
...
             pass
 
-        try:
-            if isinstance(data, unicode):  # type: ignore
-                # Don't change it, JSON and XML ElementTree are totally able
-                # to serialize correctly u'' strings
-                return data
-        except NameError:
-            return str(data)
         return str(data)
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/keyvault/vendored_sdks/azure_keyvault_securitydomain/_serialization.py:931]: Isn't `data` already a `str` per L925? @evelyn-ys 

```suggestion
        return data
```

File: src/azure-cli/azure/cli/command_modules/resource/tests/hybrid_2018_03_01/test_resource_custom.py [hunk 2]
```diff
@@ -325,8 +325,7 @@ def test_deployment_prompt_file_order(self):
         result_parameters = _process_parameters(template, parameter_list)
         missing_parameters = _find_missing_parameters(result_parameters, template)
 
-        param_file_order = ["[u'secureParam', u'boolParam', u'enumParam', u'arrayParam', u'objectParam']",
-                            "['secureParam', 'boolParam', 'enumParam', 'arrayParam', 'objectParam']"]
+        param_file_order = ["['secureParam', 'boolParam', 'enumParam', 'arrayParam', 'objectParam']"]
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/resource/tests/hybrid_2018_03_01/test_resource_custom.py:328]: There is no need to put `"['secureParam', 'boolParam', 'enumParam', 'arrayParam', 'objectParam']"` in a list anymore.

L330 should also use 

```py
self.assertTrue(str(list(results.keys())) == param_file_order)
```

@zhoxing-ms

**Review by jiasli — DISMISSED** [review]
I left some comments, but they require code logic change, which is out of the scope of this PR.

**Review by necusjz — DISMISSED** [review]

File: src/azure-cli/azure/cli/command_modules/keyvault/vendored_sdks/azure_keyvault_securitydomain/_serialization.py (hunk 1 above)
Comment (evelyn-ys) [reply src/azure-cli/azure/cli/command_modules/keyvault/vendored_sdks/azure_keyvault_securitydomain/_serialization.py:931]: I actually fully copied all the files from SDK, not sure why here's a `str()`, but I prefer not modifying files under `vendored_sdks`. BTW, this `azure-keyvault-securitydomain` vendored sdk will be removed after SDK team publish to pypi

**Review by evelyn-ys — APPROVED** [review]

**Review by kairu-ms — DISMISSED** [review]

**Comment by atombrella** [comment]
@evelyn-ys @necusjz @jiasli Could you please have another look at this? There are issues with the test runs. I think it's ready to be merged.

**Comment by atombrella** [comment]
@kairu-ms

**Review by kairu-ms — APPROVED** [review]

**Review by evelyn-ys — APPROVED** [review]

**Review by wangzelin007 — APPROVED** [review]

**Review by bebound — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]

**Review by calvinhzy — APPROVED** [review]

**Review by yanzhudd — APPROVED** [review]

**Review by jiasli — APPROVED** [review]

**Comment by jiasli** [comment]
@atombrella, Thank you very much for the contribution!

**Pull request merged**